src/main/kotlin/ch/srgssr/pillarbox/monitoring/event/model/ErrorProcessor.kt

```diff
@@ -0,0 +1,131 @@
+package ch.srgssr.pillarbox.monitoring.event.model
+
+/**
+ * A processor that identifies and categorizes error messages found in an error data node.
+ *
+ * This processor determines whether an error message corresponds to a known content restriction type
+ * and annotates the data node with an appropriate classification. If the message is unknown or missing,
+ * it flags the error as a business error.
+ */
+internal class ErrorProcessor : DataProcessor {
+  /**
+   * Process only on ERROR events.
+   */
+  override fun shouldProcess(eventName: String): Boolean = eventName == "ERROR"
+
+  /**
+   * Processes the given data node to determine the type of error based on its message.
+   *
+   * If the `message` field is empty or null, the node is marked as a business error.
+   * If the message matches a predefined content restriction category, an `error_type` field is added.
+   * Otherwise, the node is flagged as a business error.
```

Comment:
Based on the code, isn't it flagged as a business error if the message is recognized?
```suggestion
   * If the message matches a predefined content restriction category, an `error_type` field is added and the node is flagged as a business error.
```